src/test/java/eu/europeana/metis/sandbox/common/TestUtils.java

```diff
@@ -25,4 +36,59 @@ public String readFileToString(String file) throws IOException {
     }
     return inputStream.readAllBytes();
   }
+
+  public List<ByteArrayInputStream> getContentFromZipFile(InputStream providedZipFile)
+      throws IOException {
+    String zipFolder = UUID.randomUUID().toString();
+    ZipFile zipFile = this.createTempZipFile(providedZipFile, zipFolder);
+
+    List<ByteArrayInputStream> result;
+    try {
+      List<InputStream> streams = this.getContentFromZipFile(zipFile);
+      List<ByteArrayInputStream> tmpResult = new ArrayList<>(streams.size());
+      Iterator<InputStream> streamsIterator = streams.iterator();
+
+      while (true) {
```

Comment:
Using a `while(true)` loop is a bad code practice. It makes it hard to maintain the code, the only way to break the loop is with a break, and it also gives a risk of an infinite loop. 
Specially in this case, since you're iterating through each element of the `streams` list,  you can find out the size of this list and use it, which makes this `while(true)` loop is completely unnecessary.
One solution using the iterator is to add this condition to the while loop like so:
`while(streamsIterator.hasNext()){...}`
There are other ways to iterate through the list of course, I would personally try to go with a for each loop.
Either way, please change this loop

Reply:
The `while(true)` loop is still here, please change this loop